task/backend/scheduler.go

```diff
@@ -675,16 +688,21 @@ func (r *runner) fail(qr QueuedRun, runLogger *zap.Logger, stage string, reason
 func (r *runner) executeAndWait(ctx context.Context, qr QueuedRun, runLogger *zap.Logger) {
 	defer r.wg.Done()
 
+	defer func() {
```

Comment:
Undo this change to defer the call to FinishRun. The specific error messages at the specific points are necessary debugging information. Without spending time to trace through a lot of code, I can't say what happens when FinishRun happens _after_, rather than before, calls to r.fail or r.startFromWorking.

Reply:
In the new system we wont be able to update the status of a task or add to its logs after it is "finished" something that we were guilty of doing quite a bit in the scheduler. I think we can still accomplish more clear error messaging but keep the finish as a defer.